src/panels/lovelace/ha-panel-lovelace.ts

```diff
@@ -199,33 +208,67 @@ export class LovelacePanel extends LitElement {
         oldHass.floors !== this.hass.floors
       ) {
         if (this.hass.config.state === "RUNNING") {
-          this._askRefreshConfig();
+          this._debounceRegistriesChanged();
         }
       }
       // If ha started, refresh the config
       if (
         this.hass.config.state === "RUNNING" &&
         oldHass.config.state !== "RUNNING"
       ) {
-        this._refreshConfig();
+        this._regenerateStrategyConfig();
       }
     }
   }
 
-  private _askRefreshConfig = () => {
+  private _debounceRegistriesChanged = debounce(
+    () => this._registriesChanged(),
+    200
+  );
+
+  private _registriesChanged = async () => {
+    if (!this.hass || !this.lovelace) {
+      return;
+    }
+    const rawConfig = this.lovelace.rawConfig;
+
+    if (!isStrategyDashboard(rawConfig)) {
+      return;
+    }
+
+    const oldConfig = this.lovelace.config;
+    const generatedConfig = await generateLovelaceDashboardStrategy(
+      rawConfig,
+      this.hass!
+    );
+
+    const newConfig = checkLovelaceConfig(generatedConfig) as LovelaceConfig;
+
+    // Ask to regenerate if the config changed
+    if (!deepEqual(newConfig, oldConfig)) {
+      this._askRegenerateStrategyConfig();
```

Comment:
Can we pass the new config to `_askRegenerateStrategyConfig` and apply it immediately when the user confirms?

Reply:
I was thinking doing that but if the view strategy ask for a refresh we don't have the config. And in general the change impacted the view config, not the dashboard config.